### PR TITLE
Add an auto-update.sh script and Github workflow

### PR DESCRIPTION
**Pull request by daddykotex**
Add a script to do an update of the AWS sdk specifications. The workflow can be run manually, but it's meant to run on a schedule (weekly).

### CODE REVIEW

File: .github/workflows/auto-update.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: Update submodule references
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "15 12 * * 1" # At 12:15 on Monday.
```
Comment (kubukoz) [inline .github/workflows/auto-update.yml:6]: **thought:** oddly specific hour...

File: scripts/auto-update.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash 
+
+set -euo pipefail
+
+function errcho() {
```
Comment (kubukoz) [inline scripts/auto-update.sh]: **question:** is this used?

Comment (daddykotex) [reply scripts/auto-update.sh]: ohh, leftover

File: .github/workflows/auto-update.yml (hunk 1 above)
Comment (daddykotex) [reply .github/workflows/auto-update.yml:6]: no particular reason haha 🍜

**Review by lewisjkl — APPROVED** [review]

**Review by kubukoz — APPROVED** [review]

**Review by Baccata — APPROVED** [review]
Thanks a lot for this initiative, I wanted to get around to it at some point !

**Comment by daddykotex** [comment]
Unfortunately, I could not test it w/o a merge, so now, I've found an issue ... https://github.com/disneystreaming/aws-sdk-smithy-specs/pull/4

**Pull request merged**